### DISCUSSION TITLE
Thank you for the DirectML update

### DISCUSSION THREAD

**Discussion by BadWXfcst** [vladmandic/sdnext#1701] [General]
Whatever magic @lshqqytiger implemented in DirectML drastically increased the speed and VRAM availability of my old 8GB AMD GPU. Thank you.

**Comment by vladmandic** [comment]
@lshqqytiger FYI :)

**Comment by lshqqytiger** [comment]
There was no expectation about speed or vram on recent update.. interesting.

**Comment by TeutonJon78** [comment]
I'm finding the opposite on my RX580. At the same settings yesterday I could generated 512x912, but now I'm getting OOM.

Commit cdc803b127e4aa660982799186d74b491165d15f - 512x912 works fine
Commit 0a52c44e739e805cb9ed4a74260584d998a8b826 -- RuntimeError: Could not allocate tensor with 1703411712 bytes. There is not enough GPU video memory available!

And same at current head (a7f5e75a350a5e517c64174db5a6aca318e395e5), OOM

@lshqqytiger, any thoughts?

Edit: VRAM utilization is sitting at 6.3 GB instead of the normal 7.4 GB.
